extensions/amp-viewer-gpay-button/0.1/amp-viewer-gpay-button.js

```diff
@@ -14,34 +14,49 @@
  * limitations under the License.
  */
 
-import {Layout} from '../../../src/layout';
+import {AmpPaymentGoogleIntegration} from '../../../src/service/payments/amp-payment-google';
+import {getServiceForDoc} from '../../../src/service';
 
-export class AmpViewerGpayButton extends AMP.BaseElement {
+/** @const {string} */
+const TAG = 'amp-viewer-gpay-button';
+
+class AmpViewerGpayButton extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+  }
 
-    /** @private {string} */
-    this.myText_ = 'hello world';
-
-    /** @private {?Element} */
-    this.container_ = null;
+  /** @override */
+  isLayoutSupported(unusedLayout) {
+    return true;
   }
 
   /** @override */
   buildCallback() {
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    this.applyFillContent(this.container_, /* replacedContent */ true);
+    /**
+     * @private {!AmpPaymentGoogleIntegration}
+     */
+    this.paymentsIntegration_ = getServiceForDoc(
```

Comment:
Does this component need to be experiment-gated? Or is it already launched?

Reply:
the component is already launched with partners on the aog fork

Reply:
Since the component already has a validator file, there's nothing preventing users from using it in production. Generally we experiment-gate components until the component implementation and API is finalized and well documented. This guarantees that further changes to the component during implementation doesn't break existing users. If you don't anticipate further changes to your API, this is probably fine. We should try to get related / mutually dependent changes in the same release though (e.g. the validator changes referenced above).

Reply:
right, there shouldn't be any api changes from now till after the validator changes